tiledb/sm/crypto/encryption_key.h

```diff
@@ -87,11 +87,17 @@ class EncryptionKey {
       uint32_t key_length);
 
  private:
-  /** Buffer holding the encryption key. */
-  Buffer key_;
-
-  /** The encryption type. */
+   /** The encryption type. */
   EncryptionType encryption_type_;
+
+  /** Size of the array storing the encryption key. */
+  static const uint32_t max_key_length_ = 32;
```

Comment:
I suggest constants to be moved to `tiledb/sm/misc/constants.{h,cc}`.

Reply:
The maximum key length isn't a functionally independent constant, but wholly dependent on the ciphers supported. This length is an implementation byproduct rather than an independently tunable parameter.

It should, however, be declared `constexpr size_t`.

Reply:
Yeah, that makes sense, thanks for pointing it out.